packages/terminal/src/browser/terminal-frontend-contribution.ts

```diff
@@ -411,6 +416,29 @@ export class TerminalFrontendContribution implements FrontendApplicationContribu
                 (this.shell.activeWidget as TerminalWidget).scrollPageDown();
             }
         });
+        commands.registerCommand(TerminalCommands.TOOGLE_TERMINAL);
+        commands.registerHandler(TerminalCommands.TOOGLE_TERMINAL.id, {
+            execute: widget => this.toogleTerminal(widget)
+        });
+    }
+
+    protected toogleTerminal(widget: Widget): void {
+        const allTerminals = this.widgetManager.getWidgets(TERMINAL_WIDGET_FACTORY_ID) as TerminalWidget[];
```

Comment:
There is no need to get all widgets and so on, you already should have access to the `currentTerminal` or `lastUsedTerminal`:

```ts
const currentTerminal = this.currentTerminal;
const lastUsedTerminal = this.lastUsedTerminal;
```

I believe this should be enough for all use-cases.

Reply:
Okay, I will try this option then and let you know. Thanks!

Reply:
I have modified this part and it is now using `this._currentTerminal` instead of retrieving all the widgets.

However, I am still having issues with the focus, to switch it between the terminal and the editor.